.github/workflows/deploy.yml

```diff
@@ -40,34 +56,52 @@ jobs:
     outputs:
       env_matrix: ${{ steps.set-matrix.outputs.env_matrix }}
       service_matrix: ${{ steps.set-matrix.outputs.service_matrix }}
+      deploy_matrix: ${{ steps.set-matrix.outputs.deploy_matrix }}
+      nr_matrix: ${{ steps.set-matrix.outputs.nr_matrix }}
     steps:
-      - name: Generate Environment & Service Matrices
+      - name: Validate Selected Services
+        run: |
+          if [ "${{ github.event.inputs.deploy_api }}" != "true" ] && \
+             [ "${{ github.event.inputs.deploy_worker }}" != "true" ] && \
+             [ "${{ github.event.inputs.deploy_ws }}" != "true" ] && \
+             [ "${{ github.event.inputs.deploy_webhook }}" != "true" ]; then
+            echo "Error: At least one service must be selected for deployment."
+            exit 1
+          fi
+    
+      - name: Generate Environment, Service, and Deploy Matrices
         id: set-matrix
+        env:
+          WORKER_SERVICE: ${{ vars.WORKER_SERVICE }}
         run: |
           envs=()
           services=()
+          deploy_matrix=()
+          nr=()
 
           # Collect selected environments
-          if [ "${{ github.event.inputs.environment }}" == "development" ]; then
-            envs+=("\"development\"")
+          if [ "${{ github.event.inputs.environment }}" == "staging" ]; then
+            envs+=("\"staging-eu\"")
```

Comment:
we have staging-eu?

Reply:
Staging exist in EU region. Some environments name were taken up here in actions. So I made parity in nomenclature with env-location.